Gordon360/Controllers/AccountsController.cs

```diff
@@ -61,32 +61,25 @@ public async Task<ActionResult<IEnumerable<BasicInfoViewModel>>> SearchAsync(str
         {
             var accounts = await _accountService.GetAllBasicInfoExceptAlumniAsync();
 
-            var searchResults = accounts.AsParallel()
-                .Select(account => (matchKey: account.MatchSearch(searchString), account))
-                .Where(pair => pair.matchKey is not null)
-                .OrderBy(pair => pair.matchKey)
-                .Select(pair => pair.account);
+            var searchResults = _accountService.Search(searchString, accounts);
 
             return Ok(searchResults);
         }
 
         /// <summary>
         /// Return a list of accounts matching some or all of the search parameter.
         /// </summary>
-        /// <param name="firstnameSearch">The firstname portion of the search</param>
-        /// <param name="lastnameSearch">The lastname portion of the search</param>
+        /// <param name="firstName">The firstname portion of the search</param>
+        /// <param name="lastName">The lastname portion of the search</param>
         /// <returns> All accounts matching some or all of both the firstname and lastname parameters, sorted by how well the account matched the search.</returns>
         [HttpGet]
         [Route("search/{firstnameSearch}/{lastnameSearch}")]
-        public async Task<ActionResult<IEnumerable<BasicInfoViewModel>>> SearchWithSpaceAsync(string firstnameSearch, string lastnameSearch)
+        public async Task<ActionResult<IEnumerable<BasicInfoViewModel>>> SearchWithSpaceAsync(string firstName, string lastName)
```

Comment:
This is actually broken as is, the param names in the route need to match the param names in the method signature in order for ASP.NET Core to match them.